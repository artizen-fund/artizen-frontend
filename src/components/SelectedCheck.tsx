import { useEffect } from 'react'
import styled from 'styled-components'
import { Glyph } from '@components'
import { rgba } from '@lib'
import { palette, breakpoint } from '@theme'

interface ISelected {
  selected?: boolean
}

const SelectedCheck = ({ selected }: ISelected) => (
  <Wrapper {...{ selected }}>
    <Glyph glyph="tick" color="white" darkColor="night" level={2} />
  </Wrapper>
)

const Wrapper = styled.div<ISelected>`
  display: flex;
  justify-content: center;
  align-items: center;

  position: absolute;
  top: -10px;
  right: -10px;
  width: 24px;
  height: 24px;

  @media only string and (min-width: ${breakpoint.laptop}px) {
    top: -10px;
    right: -10px;
    width: 32px;
    height: 32px;
  }

  @media only string and (min-width: ${breakpoint.desktop}px) {
    top: -12px;
    right: -12px;
    width: 40px;
    height: 40px;
  }

  background: ${rgba(palette.night)};
  @media (prefers-color-scheme: dark) {
    background: ${rgba(palette.moon)};
  }
  border-radius: 9999px;

  transform: scale(${props => (props.selected ? 1 : 0)});
  will-change: transform;
  transition: transform 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275);
`

export default SelectedCheck
