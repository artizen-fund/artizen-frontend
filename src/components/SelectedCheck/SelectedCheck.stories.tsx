import SelectedCheck from './'

export default {
  title: 'components/SelectedCheck',
  component: SelectedCheck,
  argTypes: {},
}

export const SelectedCheckComponent = (props: any) => <SelectedCheck {...props} />
