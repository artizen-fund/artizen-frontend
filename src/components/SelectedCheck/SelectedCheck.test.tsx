import renderer from 'react-test-renderer'
import SelectedCheck from './'

describe('SelectedCheck', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SelectedCheck />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
