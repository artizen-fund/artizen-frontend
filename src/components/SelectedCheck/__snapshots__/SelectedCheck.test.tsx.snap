// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedCheck renders correctly 1`] = `
<div
  className="sc-d66eb548-0 dAxGdF"
>
  <div
    className="sc-3ba9b100-0 fSONTk"
    color="white"
  />
</div>
`;
